src/SettingsProvider.tsx

```diff
@@ -1,16 +1,27 @@
-import React, { ReactNode, useContext } from 'react'
+import React, { ReactNode, useContext, useEffect } from 'react'
 import useLocalStorage from 'react-use/lib/useLocalStorage'
 import useSearchParam from 'react-use/lib/useSearchParam'
 
 import { SHOW_LATEST_RCS, USE_YARN_PLUGIN } from './utils'
 import { updateURL } from './utils/update-url'
 
+interface SETTINGS {
+  [SHOW_LATEST_RCS]: boolean
+  [USE_YARN_PLUGIN]: boolean
+}
+
+const settings: SETTINGS = {
+  [SHOW_LATEST_RCS]: false,
+  [USE_YARN_PLUGIN]: false,
+}
+
+type SET_SETTINGS = (settings: SETTINGS) => void
+
+const setSettings: SET_SETTINGS = (settings: SETTINGS) => {}
+
```

Comment:
I think we can remove the global `settings`/`setSettings`

```suggestion
interface SETTINGS {
  [SHOW_LATEST_RCS]: boolean
  [USE_YARN_PLUGIN]: boolean
}

type State = {
  settings: SETTINGS
  setSettings(settings: SETTINGS): void
}
```